lib/xnvme_be_linux_hugepage.c

```diff
@@ -0,0 +1,179 @@
+// Copyright (C) Simon A. F. Lund <simon.lund@samsung.com>
+// SPDX-License-Identifier: Apache-2.0
+#include <xnvme_be.h>
+#include <xnvme_be_nosys.h>
+#ifdef XNVME_BE_LINUX_ENABLED
+#include <xnvme_dev.h>
+#include <errno.h>
+
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <sys/queue.h>
+#include <linux/limits.h>
+
+struct huge_alloc {
+	char path[PATH_MAX];
+	int fd;
+	void *addr;
+	size_t len;
+
+	SLIST_ENTRY(huge_alloc) link;
+};
+
+SLIST_HEAD(huge_alloc_slist, huge_alloc);
+struct huge_alloc_slist huge_alloc_head;
+
+size_t
+get_hugepage_size()
+{
+	char line[128];
+	FILE *fp;
+	size_t hugepage_size = 0;
+
+	fp = fopen("/proc/meminfo", "r");
+
+	while (fgets(line, 128, fp)) {
+		if (sscanf(line, "Hugepagesize: %16lu kB", &hugepage_size)) {
+			XNVME_DEBUG("Hugepage size: %lu kB", hugepage_size);
+			return hugepage_size * 1024;
+		}
+	}
+	XNVME_DEBUG("Unable to find hugepage size");
+	return 0;
+}
+
+bool
+verify_hugetlbfs_path(char *path)
+{
+	char line[1024];
```

Comment:
This seems to also be used for paths, so how about use ``PATH_MAX`` as well and then initialize with nul-byte, e.g.:
```
	char line[PATH_MAX] = { '\0' };
	char search_str[PATH_MAX] = { '\0' };
```